eliminate the difference between when having no RTT sample and having one (#3525)
quicwg/base-drafts: draft-ietf-quic-recovery.md

  min_rtt.  This limits the underestimation that a misreporting peer can cause
  to the smoothed_rtt.

smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.  When
there are no samples for a network path:

~~~
smoothed_rtt = rtt_sample
rttvar = rtt_sample / 2
~~~

where rtt_sample is the initial RTT.

On the first RTT sample for the network path, smoothed_rtt and rttvar are set to

> the values using the formula above, using that first RTT sample as rtt_sample.

Reviewer: ```suggestion
the values using the formula above, using that first RTT sample as rtt_sample.
This ensures a measurement erases the history of any persisted or default value.
```

Author: Thank you for the suggestion. Applied.

  min_rtt.  This limits the underestimation that a misreporting peer can cause
  to the smoothed_rtt.

smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.  When
there are no samples for a network path:

~~~
smoothed_rtt = rtt_sample
rttvar = rtt_sample / 2
~~~

where rtt_sample is the initial RTT.

On the first RTT sample for the network path, smoothed_rtt and rttvar are set to
the values using the formula above, using that first RTT sample as rtt_sample.

> This ensures a measurement erases the history of any persisted or default value.

Reviewer 2: ```suggestion
This ensures that the first measurement erases the history of any persisted or default value.
```

Author: Thank you for the suggestion. Applied in c4ed9aa.

  min_rtt.  This limits the underestimation that a misreporting peer can cause
  to the smoothed_rtt.

smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.  When

> there are no samples for a network path:

Reviewer 2: ```suggestion
smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.

When there are no samples for a network path, and on the first RTT sample for
the network path:
```

Author: Splitting the text when having no samples and having one is intentional, see the changes suggested in https://github.com/quicwg/base-drafts/pull/3525#pullrequestreview-374825664.

I'm personally fine with the changes proposed here, but I would like to see what @ianswett thinks. Maybe it's OK, as the we have two paragraphs right after, discussing about each of the two conditions.

Reviewer: I think @janaiyengar suggestion seems good and slightly clearer than before, though maybe line 361("where rtt_sample is the initial RTT." needs to be updated too?

~~~
smoothed_rtt = rtt_sample
rttvar = rtt_sample / 2
~~~

where rtt_sample is the initial RTT.

On the first RTT sample for the network path, smoothed_rtt and rttvar are set to
the values using the formula above, using that first RTT sample as rtt_sample.
This ensures that the first measurement erases the history of any persisted or

> default value.

Reviewer 2: ```suggestion
Before any RTT samples are available, the initial RTT is used as rtt_sample. 
On the first RTT sample for the network path, that sample is used as rtt_sample.
This ensures that the first measurement erases the history of any persisted or
default values.
```

On the first RTT sample for a network path, the smoothed_rtt is set to the

> latest_rtt.

Reviewer: Moving this down above "On subsequent RTT samples," and adding that rttvar is also set on the first RTT sample  would help clarify this.

smoothed_rtt and rttvar are computed as follows, similar to {{?RFC6298}}.  When
there is less than one sample for a network path:

~~~
smoothed_rtt = rtt_sample
rttvar = rtt_sample / 2
~~~

where rtt_sample is either the initial RTT when there is no sample, or the
latest RTT when there is one.

> On subsequent RTT samples, smoothed_rtt and rttvar evolve as follows:

Reviewer: I'm concerned this section is now potentially ambiguous.  I believe we want to maintain the property that the first actual RTT measurement replaces the initialized values, and I don't think that's clear enough here.

### Handshakes and New Paths

When no previous RTT is available, the initial RTT SHOULD be set to 333ms,
resulting in a 1 second initial timeout as recommended in {{?RFC6298}}. Resumed
connections over the same network SHOULD use the previous connection's final

> smoothed RTT value as the resumed connection's initial RTT.

Reviewer 2: ```suggestion
Resumed
connections over the same network SHOULD use the previous connection's final
smoothed RTT value as the resumed connection's initial RTT. When no previous
RTT is available, the initial RTT SHOULD be set to 333ms, resulting in a 1 second
initial timeout, as recommended in {{?RFC6298}}.
```

### Handshakes and New Paths

> Resumed

Reviewer: nit: reflow the text
